src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -503,6 +503,43 @@ internal static IEnumerable<string> GetFrameworkAssembliesToTreatAsUserAssemblie
 			return acw_map;
 		}
 
+		public static Dictionary<string, HashSet<string>> LoadMapFile (IBuildEngine4 engine, string mapFile)
+		{
+			var cachedMap = (Dictionary<string, HashSet<string>>)engine?.GetRegisteredTaskObject (mapFile, RegisteredTaskObjectLifetime.Build);
+			if (cachedMap != null)
+				return cachedMap;
+			var map = new Dictionary<string, HashSet<string>> ();
+			if (!File.Exists (mapFile))
+				return map;
+			foreach (var s in File.ReadLines (mapFile)) {
+				var items = s.Split (';');
```

Comment:
Don't use `string.Split(char[])`; use `string.Split(char[], int`:

```csharp
var items = s.Split (new[]{';'}, 2);
```

This way if the filesystem path contains a `;` it'll behave sanely.